src/main/java/codechicken/multipart/api/part/render/PartBakedModelRenderer.java

```diff
@@ -32,20 +32,23 @@ static <T extends ModelRenderPart> PartBakedModelRenderer<T> simple() {
     }
 
     @Override
-    default boolean renderStatic(T part, @Nullable RenderType layer, CCRenderState ccrs) {
-        if (!part.canRenderInLayer(layer)) return false;
+    default void renderStatic(T part, @Nullable RenderType layer, CCRenderState ccrs) {
 
         BlockRenderDispatcher blockRenderer = Minecraft.getInstance().getBlockRenderer();
-        Random randy = new Random();
-        return blockRenderer.renderBatched(
+        BlockState state = part.getCurrentState();
+        RandomSource randy = RandomSource.create();
+
+        if (!blockRenderer.getBlockModel(state).getRenderTypes(state, randy, part.getModelData()).contains(layer)) return;
```

Comment:
Do we need to do this check? Won't `renderBatched` already do this for us?

Reply:
Looks like `renderBatched()` will fetch the block model and immediately try to render every quad. The `getQuads()` method is `renderType` sensitive though, so depending on the block model implementation, it will return only quads compatible with the given render layer. I'll need to double check but probably this check can be removed

Reply:
Turns out that this check is required (unless there is an issue elsewhere). Example, instances of `SimpleBakedModel` always return all quads through `getQuads(...)` even if the layer passed is not in the set returned by `getRenderTypes()`. This is a default behavior inherited from `IForgeBakedModel`.

So I guess you must never call `renderBatched()` until *after* you check the layer.

Reply:
Damn, alright.